### PR TITLE
Add scripts for MAGMA builds on spock and crusher

### PR DESCRIPTION
**Pull request by mewall**

### CODE REVIEW

**Review by nicolasbock — APPROVED** [review]
:+1:

**Review by cnegre — APPROVED** [review]

**Comment by jmohdyusof** [comment]
It seems we have multiple yes/no flags (and will have more) but it isn't clear if it will cause problems to have multiple flags enabled at once? Does it make sense to have a flag that selects one of the options instead?

**Comment by mewall** [comment]
I think it makes sense but this commit doesn't introduce any new flags so I think can be merged.


The rocsolver branch introduces a BML_ROCSOLVER flag, similar to the BML_CUSOLVER flag, I will look at that logic again before submitting a PR (after this one is merged).

________________________________
From: Jamal Mohd-Yusof ***@***.***>
Sent: Friday, February 4, 2022 10:27:19 AM
To: lanl/bml
Cc: Wall, Michael E; Author
Subject: [EXTERNAL] Re: [lanl/bml] Add scripts for MAGMA builds on spock and crusher (PR #585)


It seems we have multiple yes/no flags (and will have more) but it isn't clear if it will cause problems to have multiple flags enabled at once? Does it make sense to have a flag that selects one of the options instead?

**Comment by jeanlucf22** [comment]
I don't mind it being merged. I was just making a point that we should better use our build script, in particular as users may want to see good example of how to build the code.

File: scripts/build_crusher_magma_gcc.sh
Comment (mewall) [reply scripts/build_crusher_magma_gcc.sh:22]: I can do this. My comment below was in response to Jamal's point about the flags.

**Comment by mewall** [comment]
> BTW, after PR #583 is merged, you should just need
> export BLAS_VENDOR=${BLAS_VENDOR:=OpenBLAS}
> to specify which BLAS to use

The suggested 

export BLAS_LIBRARIES=${BLAS_LIBRARIES:=$OLCF_OPENBLAS_ROOT/lib/libopenblas.a}

didn't work immediately. Rather than troubleshooting now, I'll wait until PR #583 is merged and try again using the new cmake file.

**Review by suemni — APPROVED** [review]

**Comment by nicolasbock** [comment]
@mewall I just merged lanl/bml#583.

**Review by jeanlucf22 — APPROVED** [review]

**Review by nicolasbock — APPROVED** [review]
:+1:

**Pull request merged**
